pkgs/applications/science/math/sage/pybrial.nix

```diff
@@ -0,0 +1,29 @@
+{ stdenv
+, fetchFromGitHub
+, buildPythonPackage
+}:
+# This has a cyclic dependency with sage. I don't include sage in the
+# buildInputs and let python figure it out at runtime. Because of this,
+# I don't include the package in the main nipxkgs tree. It wouldn't be useful
+# outside of sage anyways (as you could just directly depend on sage and use
+# it).
+buildPythonPackage rec {
+    pname = "pyBRiAl";
+    version = "1.2.3";
+
+    # included with BRiAl source
+    src = fetchFromGitHub {
+      owner = "BRiAl";
+      repo = "BRiAl";
+      rev = "${version}";
+      sha256 = "0qy4cwy7qrk4zg151cmws5cglaa866z461cnj9wdnalabs7v7qbg";
+    };
+
+    preConfigure = "cd sage-brial";
```

Comment:
is `sourceRoot` an option here?

Reply:
As a small rant: I don't like that you have to explicitly specify the name the "real" source root will have after unpacking when using `sourceRoot`. For example in this case I have to specify `sourceRoot = "source/sage-brial"`. I'd prefer if the logic that is used to determine the `sourceRoot` would always be executed and just rename the `sourceRoot` to something standard, like `src`. I think ArchLinux does it that way. Than it can just be optional weather or not to additionally `cd` into `src`.

But yes, `sourceRoot` is the better option here. I don't remember why I didn't use it, if I even had a conscious reason.